lib/runner.js

```diff
@@ -52,11 +58,45 @@ class Runner extends Emittery {
 			if (!scheduledStart) {
 				scheduledStart = true;
 				process.nextTick(() => {
+					if (lastMetadata) {
+						const includedInRange = this.lines.some(n => n >= lastCallsiteLineNumber);
+						lastMetadata.exclusive = this.match.length === 0 ?
+							includedInRange :
+							includedInRange && lastMetadata.exclusive;
+					}
+
 					hasStarted = true;
 					this.start();
 				});
 			}
 
+			if (this.lines.length > 0) {
+				this.runOnlyExclusive = true;
+				let callsiteLineNumber;
+
+				for (const jsCallsite of callsites()) {
+					const callsite = sourceMapSupport.wrapCallSite(jsCallsite);
+					const fileName = callsite.getFileName();
+					if (fileName === this.file) {
+						callsiteLineNumber = callsite.getLineNumber();
+						break;
+					}
+				}
+
+				if (lastMetadata) {
+					// `callsiteLineNumber` is the beginning of the current test, -1 to
+					// get the last line of the previous test.
```

Comment:
This does include whitespace, but I think that's OK for now.

Reply:
I don't think there is a way to handle this case unless you parse the source file to it's AST and then extract when the function call actually ends (this information is not in the callsite data.)